rust/noosphere-ns/Cargo.toml

```diff
@@ -37,9 +38,27 @@ tokio = { version = "1.15", features = ["io-util", "io-std", "sync", "macros", "
 libp2p = { version = "0.49.0", default-features = false, features = [ "identify", "dns", "tcp", "tokio", "noise", "mplex", "yamux", "kad" ] }
 noosphere-storage = { version = "0.1.0", path = "../noosphere-storage" }
 noosphere-core = { version = "0.2.0", path = "../noosphere-core" }
+noosphere = { version = "0.2.0", path = "../noosphere", optional = true }
+clap = { version = "^4", features = ["derive"], optional = true }
+home = { version = "~0.5", optional = true }
+toml = { version = "~0.5", optional = true }
 
 [dev-dependencies]
+
+[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies]
 rand = { version = "0.8.5" }
 test-log = { version = "0.2.11", default-features = false, features = ["trace"] }
 tracing-subscriber = { version = "~0.3", features = ["env-filter"] }
 libipld-cbor = "~0.14"
+tempdir = { version = "~0.3" }
+
+[features]
+default = []
+bootstrap = ["clap", "noosphere", "home", "toml"]
+
+[target.'cfg(not(target_arch = "wasm32"))'.features]
+default = ["bootstrap"]
```

Comment:
rationale for this being a default feature is integration into larger test suite. Otherwise we'd need to flag --features="bootstrap" everywhere, preferring for library usage to disallow this. WDYT?

Separately, this is cfg-flagged because otherwise the web builds try to build the binary, even with the main file having `#![cfg(not(target_arch = "wasm32"))]`

Reply:
As with the CLI, I think you may end up being happier allowing this to be built for `wasm32-unknown-unknown` but stubbing out the main in that case. The reason is: you will probably want to test your underlying library, so you have to stub that part out _anyway_.

Reply:
> The reason is: you will probably want to test your underlying library, so you have to stub that part out anyway.

as in, test the bootstrap library on wasm32-unknown-unknown?

(I'll take a look at the CLI's cargo config)

Reply:
No, I mean: no matter what you are going to have to deal with this command running:

```sh
cargo run --target wasm32-unknown-unknown
```

Which means even if you gate the _binary_ on a feature, you still have tests invoking the library code behind that binary (which will have to be conditioned not to run on `wasm32` arch anyway).